feat(html): Add meta http-equiv attr page (#40043)
mdn/content: files/en-us/web/html/reference/elements/meta/http-equiv/index.md

---
title: <meta> http-equiv attribute
short-title: <meta> http-equiv
slug: Web/HTML/Reference/Elements/meta/http-equiv
page-type: html-attribute
browser-compat: html.elements.meta.http-equiv
---

{{HTMLSidebar}}

> The **`http-equiv`** attribute of the {{htmlelement("meta")}} element allows you to provide pragma directives in document metadata, which are instructions for the browser that could otherwise be given by an HTTP header.

Reviewer: Pragmas don't exist in JavaScript so there is a reasonable argument to saying that "pragma directive" needs to be  explained. I think it could be omitted entirely. I used one approach here https://github.com/mdn/content/pull/40043/files#r2162735420 which could be copied. Or following this pattern

```suggestion
The **`http-equiv`** attribute of the {{htmlelement("meta")}} element allows you to provide instructions for the browser that could otherwise be given by an HTTP header.
```

Author: I like it, thanks, we mention in the next sentence that it's document-level

The metadata is _document-level metadata_ that applies to the whole page.

When a `<meta>` element has a `http-equiv` attribute, a [`content`](/en-US/docs/Web/HTML/Reference/Attributes/content) attribute defines the corresponding `http-equiv` value.
For example, the following `<meta>` tag tells the browser to refresh the page after 5 minutes:

```html
<meta http-equiv="Refresh" content="300" />
```

## Value

> The attribute's name is short for `http-equivalent` because the allowed values are names of equivalent HTTP headers.

Reviewer: I would move this back up into the top section.

> The attribute's name is short for `http-equivalent` because the allowed values are names of equivalent HTTP headers.
> The HTML specification defines the following set of `http-equiv` values:

Reviewer: The attribute's name is `http-equiv` because the allowed values are names of equivalent HTTP headers - the "short for" thing doesn't follow. I'd mention the "short for" up in the top line of the description.

But this is an important point - only a subset of headers are allowed. The fact that this is what the spec says is implicit for anything not marked non-standard. How about

```suggestion
Only a subset of the HTTP headers are supported as `http-equiv` values.
These include:
```

Author: I think it's a little looser, so the HTML spec has some values with defined behavior, but I think it's possible to simulate many others.

- https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/X-DNS-Prefetch-Control#turning_on_and_off_prefetching

Reporting and even client hints:

```html
<meta
      http-equiv="Reporting-Endpoints"
      content="csp-endpoint='https://example.com/csp-reports'"  …

<meta http-equiv="Accept-CH"…
```

I think implementations are really flex with what they will process, it may be possible to set nearly everything in this way.

Reviewer: @bsmth Thanks - interesting!

Agreed, that does make it relevant to mention the specification rather than "allowed headers".
But we should add a note capturing that browsers may offer non-standard support additional headers, and the risks associated with using these rather than the actual headers.

I also still agree with my comment here:

> The attribute's name is http-equiv because the allowed values are names of equivalent HTTP headers - the "short for" thing doesn't follow. I'd mention the "short for" up in the top line of the description.

    - The number of seconds until the page should be reloaded if the `content` attribute is a non-negative integer.
    - The number of seconds until the page should redirect to another URL if the `content` attribute is a non-negative integer followed by `;url=` and a valid URL.

    The timer starts when the page is _completely loaded_, which is after the {{domxref("Window/load_event", "load")}} and {{domxref("Window/pageshow_event", "pageshow")}} events have both fired.
    See [Accessibility concerns](#accessibility_concerns) for more information.

- `set-cookie` {{deprecated_inline}}
  - : Sets a cookie for the document.
    Browsers now ignore this pragma; use the {{httpheader("Set-Cookie")}} HTTP response header or [`document.cookie`](/en-US/docs/Web/API/Document/cookie) instead.
- `x-ua-compatible` {{deprecated_inline}}
  - : Used by legacy versions of the now-retired {{glossary("Microsoft Internet Explorer")}} so that it more closely followed specified behavior.
    If specified, the `content` attribute must have the value `"IE=edge"`.
    User agents now ignore this pragma.
    The name derives from the `X-UA-Compatible` HTTP header.

> [!WARNING]

> > Unrecognized headers or invalid values are ignored by the browser.
> > **Do not set other security headers** using `<meta http-equiv=`, as this can lead to a false sense of security.
> > Some browsers may process headers that are not listed above, but this behavior varies across implementations.

Reviewer: It's a nit, but at the point you get to the second line you don't know that you can process other headers (third line), so it doesn't make sense.

Shifting a little

```suggestion
> Some browsers process additional headers that are not listed above.
> Since unrecognized headers or invalid values are ignored, this can lead to inconsistent behavior across browser implementations.
> In particular, **Do not set other security headers** using `<meta http-equiv=`, as this can lead to a false sense of security!
```
